src/memtable.c

```diff
@@ -331,7 +333,10 @@ memtable_context_destroy(platform_heap_id hid, memtable_context *ctxt)
    ref = allocator_dec_ref(al, ctxt->insert_lock_addr, PAGE_TYPE_LOCK_NO_DATA);
    platform_assert(ref == AL_FREE);
 
-   platform_free(hid, ctxt);
+   platform_memfrag  memfrag_ctxt;
+   platform_memfrag *mf = &memfrag_ctxt;
+   memfrag_init_size(mf, ctxt, ctxt->mt_ctxt_size);
+   platform_free(hid, mf);
```

Comment:
NOTE: This does look like verbose multi-line repeat code.

@rtjohnso - I did consider whether to add a packaged macro, say, `memfrag_init()`, to which you supply the addr / init. Inside the body, we can declare a hidden structures and `platform_memfrag *mf`, do the setup, and pass-it as the 2nd arg.

Can be done ... probably .. did not try it too hard. Wanted to get this into review, and then I expect anyway to get comments on this approach.

We can re-discuss the coding impact this approach has ... and re-visit during review.